### PR TITLE
Replace TriaObject

### PR DESCRIPTION
**Pull request by peterrum**
This PR removes the old `TriaObject`, replaces its usage by the use of `std::initializer_list`, and renames `TriaObjectView` to `TriaObject`.

@bangerth Thanks for working on this! You were already at 95%!

### CODE REVIEW

File: include/deal.II/grid/tria_accessor.h [hunk 1]
```diff
@@ -1627,7 +1626,13 @@ class TriaAccessor : public TriaAccessorBase<structdim, dim, spacedim>
...
    */
   void
-  set(const dealii::internal::TriangulationImplementation::TriaObject &o) const;
+  set(const std::initializer_list<int> &o) const;
+
+  /**
+   * The same as above but for `unsigned int`.
+   */
+  void
+  set(const std::initializer_list<unsigned int> &o) const;
```
Comment (bangerth) [inline include/deal.II/grid/tria_accessor.h:1635]: Ah, nice trick. I really struggled with all of the narrowing warnings and what to do about it :-)

**Comment by bangerth** [comment]
Thanks for finishing this -- I just ran out of time last night :-)

File: include/deal.II/grid/tria_accessor.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/grid/tria_accessor.h:1635]: I really hate this (and the previous) hack. But, this was the approach to get as few narrowing warnings as possible.

**Comment by bangerth** [comment]
We can address this in a follow-up at some point where we just replace every signed integer by an unsigned one.

**Pull request merged**